core/src/main/scala/io/bullet/borer/Reader.scala

```diff
@@ -128,10 +128,17 @@ final class InputReader[Config <: Reader.Config](
   @inline def hasByte(value: Byte): Boolean     = hasByte && receptacle.intValue == value.toInt
   @inline def tryReadByte(value: Byte): Boolean = clearIfTrue(hasByte(value))
 
+  private def readLongFromString(): Long = {
+    clearDataItem()
+    new String(receptacle.charBufValue, 0, receptacle.intValue).toLong
+  }
+
   def readShort(): Short =
     if (hasShort) {
       clearDataItem()
       receptacle.intValue.toShort
+    } else if (hasChars) {
```

Comment:
> If we want to enable this "read-simple-types-from-strings" feature

I would prefer enabling this for map keys only, but as I wrote, I have no idea how to check if I am parsing a map key or not. Do you see some way for that?

Reply:
> rock-solid error handling

Is there any parsing with error handling ready for individual numeric types I could use here, or should I use `readLongFromString` as I do now and check for the range?

Reply:
Hmm... what's your reasoning behind only allowing this auto-conversion on map keys and not other elements as well? What's so special about map keys? To me they are are simply data elements like every other one as well.

The `Reader.Config` interface already has two other config flags called `readIntegersAlsoAsFloatingPoint` and `readDoubleAlsoAsFloat`. We can simply add `readStringsAlsoAsPrimitives`, which would enable this auto-conversion everywhere (but the default should be `false`).

(And while we are at it, breaking binary compatibility, we can also rename `readDoubleAlsoAsFloat` to `readDoublesAlsoAsFloat` for better consistency.)

> Is there any parsing with error handling ready for individual numeric types I could use here, or should I use readLongFromString as I do now and check for the range?

I would simply let `java.lang.Integer.parseInteger`, `java.lang.Short.parseShort`, `java.lang.Float.parseFloat`, etc. do the job and wrap all exceptions in a `Borer.InvalidInputData`.
When reading booleans from a String I would also allow for "off" and "no" to be read as `false` and "on" and "yes" to be read as `true`, in addition to "false" and "true".

Reply:
I am not thinking about it as auto-conversion, more like map key encoding / decoding - which is a topic of this PR. I have implemented the encoding, that I think is easy and straightforward. I need some decoding as well. A universal auto-conversion is a possible way, but as it achieves more than desired, it raises the need of config flags.

Reply:
I don't quite get the point of your last comment.
What is it that you are suggesting?

Reply:
I am just trying to explain why I wanted to limit the string parsing to map key. I have no idea how to achieve that, therefore I am unable to suggest something. I leave the decision to you - if you think a broader implementation accepting strings as primitives everywhere is good with you, I can proceed.

There is only one minor point I do not like about this: it will be possible to encode a map to JSON without any flags, the numeric keys will be encoded as string, but when one wants to decode the result of such encoding, it will be necessary to use the config allowing `readStringsAlsoAsPrimitives`. If you think this is fine, I can live with that.

Reply:
The `Reader` abstraction has no way of knowing what the role of the next data element is. It doesn't have to know, because in CBOR all data element types can appear in all "positions". It's only JSON that limits certain positions (map keys) to certain types (Strings).
As such there is no readily available way to restrict the "primitives-from-string" auto-conversion to map keys at the reader level and I wouldn't like to change the design or add additional layers/complexity just for this (minor, IMHO) feature.
So, I'm afraid it's going to be an all-or-nothing decision for or against the `readStringsAlsoAsPrimitives` functionality.

> There is only one minor point...

Yes, I agree. This isn't perfect but acceptable to me. _borer_ offers configuration options at the encoding or decoding level, not overarching, across everything. A certain amount of asymmetry is ok, I think.
We already have `readIntegersAlsoAsFloatingPoint` and `readDoubleAlsoAsFloat`, which are also somewaht asymmetric.